deploy/traits.yaml

```diff
@@ -833,19 +842,39 @@ traits:
   - name: tls-termination
     type: string
     description: The TLS termination type, like `edge`, `passthrough` or `reencrypt`.Refer
-      to the OpenShift documentation for additional information.
+      to the OpenShift route documentation for additional information.
   - name: tls-certificate
     type: string
-    description: The TLS certificate contents.Refer to the OpenShift documentation
+    description: The TLS certificate contents.Refer to the OpenShift route documentation
       for additional information.
+  - name: tls-certificate-secret
+    type: string
+    description: The TLS certificate contents read from a secret. The format is "secret-name[/key-name]",
```

Comment:
If I understand it correctly, this is not the content read from a Secret. That is the reference of the secret name and key containing the TLS certificate.

Reply:
The certificate is stored in the `secret`, isn't the same thing for configuration purposes ? The user wants to setup the certificate to be sourced either from a `secret`  or directly from CLI.

Reply:
The way it is put, we may understand that the property must be the content of the TLS certificate, while it must be the secret containing the certificate. I find the later a lot more descriptive as to what is expected for the end user.